orc8r/gateway/python/magma/ctraced/trace_manager.py

```diff
@@ -0,0 +1,117 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import logging
+import errno
+import os
+import pathlib
+import subprocess
+from subprocess import SubprocessError
+from .command_builder import get_trace_builder
+
+_TRACE_FILE_NAME = "call_trace.pcap"
+_MAX_FILESIZE = 4000  # ~ 4 MiB for a trace
+
+
+class TraceManager:
+    """
+    TraceManager is a wrapper for tshark/tcpdump specifically for starting and
+    stopping call/interface/subscriber traces.
+
+    Only a single trace can be captured at a time.
+    """
+    def __init__(self, config):
+        self._is_active = False # is call trace being captured
+        self._proc = None
+        self._trace_directory = config.get("trace_directory",
+                                           "/var/opt/magma/trace")  # type: str
+        # Specify southbound interfaces
+        self._trace_interfaces = config.get("trace_interfaces",
+                                           ["eth0"])  # type: List[str]
+        self._trace_filename = self._trace_directory + "/" + _TRACE_FILE_NAME
+
+        tool_name = config.get("trace_tool", "tshark")  # type: str
+        self._trace_builder = get_trace_builder(tool_name)
+
+    def start_trace(self) -> bool:
+        """Start a call trace.
+
+        Captures all packets across the eth0 interface.
+
+        Returns:
+            True if successfully started call trace
+        """
+        if self._is_active:
+            logging.error("Failed to start trace: Trace already active")
+            return False
+
+        command = self._trace_builder.build_trace_command(
+            self._trace_interfaces,
+            _MAX_FILESIZE,
+            self._trace_filename)
```

Comment:
minor nit: can you add some identifier like timestamp to this file, this would help in debugging.

Reply:
Sure